src/core/ext/transport/inproc/inproc_transport.cc

```diff
@@ -1103,10 +1087,6 @@ void perform_stream_op(grpc_transport* gt, grpc_stream* gs,
                error);
     grpc_core::ExecCtx::Run(DEBUG_LOCATION, on_complete, GRPC_ERROR_REF(error));
   }
-  if (needs_close) {
```

Comment:
why is this not needed anymore?

Reply:
This turned out to be dead code. If you look at this particular function, needs_close starts as false and never gets set to true in any circumstance. I don't know why the compiler never identified it as unreachable code (could be because the code overall is quite complicated?)